exporter/prometheusremotewriteexporter/helper.go

```diff
@@ -38,16 +38,62 @@ func batchTimeSeries(tsMap map[string]*prompb.TimeSeries, maxBatchByteSize int)
 	}
 
 	if len(tsArray) != 0 {
-		wrapped := convertTimeseriesToRequest(tsArray)
+		wrapped := convertTimeseriesToRequest(tsArray, nil)
+		requests = append(requests, wrapped)
+	}
+
+	mArray := make([]prompb.MetricMetadata, 0, len(m))
+	sizeOfCurrentBatch = 0
+	i = 0
+	for _, v := range m {
+		sizeOfM := v.Size()
+
+		if sizeOfCurrentBatch+sizeOfM >= maxBatchByteSize {
+			wrapped := convertTimeseriesToRequest(nil, mArray)
```

Comment:
should this just be convertMetadataToRequest?  Seems odd to always call it with 1 argument nil.

Reply:
Separate method created.